package/MDAnalysis/core/_get_readers.py

```diff
@@ -162,10 +162,10 @@ def get_writer_for(filename, format=None, multiframe=None):
     """
     if filename is None:
         format = 'NULL'
-    elif format is None:
+    elif format in ('', None):
```

Comment:
It's better to just raise an error for `format == ""` – I don't like making "" equivalent to `None` (which also changes the API). Just do
```python
if format == "":
   raise ValueError("format cannot be the empty string --- only None or valid formats are supported")
```